bit_matrix.go

```diff
@@ -0,0 +1,100 @@
+package rsmt2d
+
+import "fmt"
+
+type bitMatrix struct {
```

Comment:
Maybe, I should note that the thing is not go-rountine-safe. That doesn't matter at all right now but I know that @evan-forbes experimented with paralleization in https://github.com/lazyledger/rsmt2d/issues/5 already.

Reply:
What's the alternative here? How much work would it be to make this thread-safe? Is there not a standard library for this that is thread-safe?

Reply:
> How much work would it be to make this thread-safe?

Basically adding mutexes for accessing the underlying field. 

> Is there not a standard library for this that is thread-safe?

Not to my knowledge.

Also, we currently do not need this to be thread-safe as we aren't accessing any of this via go-routines.